paddle/gserver/gradientmachines/MultiGradientMachine.cpp

```diff
@@ -171,6 +171,12 @@ MultiGradientMachine::MultiGradientMachine(const ModelConfig& config,
   }
 }
 
+MultiGradientMachine::~MultiGradientMachine() {
```

Comment:
Maybe in destructor, we should check whether MultiGradientMachine is `finished` or not?

If MultiGradientMachine is not finish, then invoke `this->finish();`